src/ng-select/items-list.ts

```diff
@@ -94,8 +101,30 @@ export class ItemsList {
     }
 
     filter(term: string) {
-        const filterFuncVal = this._getDefaultFilterFunc(term);
-        this._filteredItems = term ? this._items.filter(val => filterFuncVal(val)) : this._items;
+        if (!term) {
+            this._filteredItems = this._items;
+            return;
+        }
+
+        this._filteredItems = [];
+        term = searchHelper.stripSpecialChars(term).toUpperCase();
```

Comment:
as I have looked, `toLowerCase` is much faster.
https://jsperf.com/javascript-touppercase-vs-tolowercase/2